### PR TITLE
6-get-api-review-id

### PR DESCRIPTION
**Pull request by tyr-ros**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -183,4 +183,53 @@ describe('GET: /api/reviews', () => {
...
+
+                })
+        })
+        it("returns a 404 if given an invalid id", () => {
+            return request(app)
+                .get('/api/reviews/cat/comments')
+                .expect(404)
+                .then(({ body }) => {
+                    expect(body.msg).toBe('Not found')
+
```
Comment (Jodiech) [inline __tests__/app.test.js:230]: This test should be a status 400 - for an invalid ID 😄

File: __tests__/app.test.js [hunk 2]
```diff
@@ -183,4 +183,53 @@ describe('GET: /api/reviews', () => {
...
+                    body: expect.any(String),
+                    votes: expect.any(Number),
+                    author: expect.any(String),
+                    created_at: expect.any(String)
+                };
+                body.forEach(comment => {
+                    expect(comment).toMatchObject(commentProperties);
+                });
+            })
+    })
```
Comment (Jodiech) [inline __tests__/app.test.js:212]: I'd like to see another test here for a - Status 200, valid ID, but has no comments and responds with an empty array of comments 👍

File: controllers/comments-controllers.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const { fetchCommentsByReviewId } = require('../models/comments-models')
+
+exports.getCommentsByReviewId = (req, res, next) => {
+    const { review_id } = req.params;
+    fetchCommentsByReviewId(review_id).then((comments) => {
+        res.status(200).send({ 'comments': comments })
```
Comment (Jodiech) [inline controllers/comments-controllers.js:6]: Good to see you are returning the data back with a comments key 😄

File: models/comments-models.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+exports.fetchCommentsByReviewId = (review_id) => {
+    return db
+        .query(`SELECT comment_id, votes, created_at, author, body, review_id
+        FROM comments
+        WHERE review_id = $1
+        ORDER BY created_at DESC;
+        `, [review_id])
+        .then((result) => {
+            if (result.rows.length === 0) {
+                return Promise.reject({ status: 404 })
```
Comment (Jodiech) [inline models/comments-models.js:12]: Great work with this query! 👍

**Review by Jodiech — COMMENTED** [review]
Nice work on this ticket! I have left a few comments, mainly on adding another 200 test (please see comments below) please submit another PR when you have made the changes 🚀

File: __tests__/app.test.js [hunk 5]
```diff
@@ -191,6 +191,16 @@ describe('GET: /api/reviews/:review_id/comments', () => {
...
             .expect(200)
     })
+    it('responds with status code 200 and a blank array if given a valid id with no reviews', () => {
+        return request(app)
+            .get('/api/reviews/1/comments')
+            .expect(200)
+            .then(( res ) => {
+                const body = res.body.comments
+                expect(body).toBeInstanceOf(Array)
+
```
Comment (MrMcSnail) [inline __tests__/app.test.js:201]: Make sure you specify that its really an empty array. Totally okay to hard-code an empty array and assert that the returned one is equal to it.

**Review by MrMcSnail — CHANGES_REQUESTED** [review]
Make that one change them DM me (Kyle) and I'll look at it personally before the end of the day

**Review by MrMcSnail — APPROVED** [review]
🥳 🧨 🎆 🍾
